Source/santad/SNTEndpointSecurityManager.m

```diff
@@ -0,0 +1,159 @@
+/// Copyright 2019 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#import "Source/santad/SNTEndpointSecurityManager.h"
+
+#import "Source/common/SNTLogging.h"
+
+#include <EndpointSecurity/EndpointSecurity.h>
+
+#include <bsm/libbsm.h>
+
+@interface SNTEndpointSecurityManager ()
+
+@property(nonatomic) es_client_t *client;
+@property (nonatomic, copy) void (^decisionCallback)(santa_message_t);
+@property (nonatomic, copy) void (^logCallback)(santa_message_t);
+
+@end
+
+@implementation SNTEndpointSecurityManager
+
+- (instancetype)init API_AVAILABLE(macos(10.15)) {
+  self = [super init];
+  if (self) {
+    es_client_t *client = NULL;
+    es_new_client_result_t ret = es_new_client(&client, ^(es_client_t *c, const es_message_t *m) {
```

Comment:
I'm thinking that the content of this block should probably be moved into a method/function and the block just calls it.